pkg/executables/clusterctl.go

```diff
@@ -64,7 +64,7 @@ func imageRepository(image v1alpha1.Image) string {
 // used by cluster api to install components.
 // See: https://cluster-api.sigs.k8s.io/clusterctl/configuration.html
 func (c *Clusterctl) buildOverridesLayer(clusterSpec *cluster.Spec, clusterName string, provider providers.Provider) error {
-	versionsBundle := clusterSpec.RootVersionsBundle()
+	versionsBundle := &clusterSpec.Bundles.Spec.VersionsBundles[0]
```

Comment:
yeah more examples than the input should probably be a VersionsBundle
Or maybe you can create a new struct `ManagementComponents` and build it from a Bundles

```
package cluster
func ManagementComponentsFromBundles(bundles) ManagementComponents {
  return newManagementComponents(bundles.Spec.VersionsBundles[0])
}
```

Reply:
Did you mean add a VersionsBundle input here to the helper or the clusterctl Upgrade interface?

E.g. 

```
```

```
if err := c.buildOverridesLayer(cluster.ManagementComponentsVersionsBundle(clusterSpec), clusterSpec, clusterName, provider); err != nil {
    return nil, err
}
```

Reply:
Discussed offline. This one will be a bit more of a lift, and we can make this change when splitting the PR